File: software/examples/delphi/ExampleCallback.pas
program ExampleCallback;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletJoystick;

type
  TExample = class
  private
    ipcon: TIPConnection;
    js: TBrickletJoystick;
  public
    procedure PressedCB(sender: TBrickletJoystick);
    procedure ReleasedCB(sender: TBrickletJoystick);
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = '9Bw'; { Change to your UID }

var
  e: TExample;

{ Callback function for pressed and released events }
procedure TExample.PressedCB(sender: TBrickletJoystick);
begin
  WriteLn('Pressed');
end;

procedure TExample.ReleasedCB(sender: TBrickletJoystick);
begin
  WriteLn('Released');
end;

procedure TExample.Execute;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  js := TBrickletJoystick.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Register callbacks for pressed and released events }
  js.OnPressed := {$ifdef FPC}@{$endif}PressedCB;
  js.OnReleased := {$ifdef FPC}@{$endif}ReleasedCB;

  WriteLn('Press key to exit');
  ReadLn;
  ipcon.Destroy;
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.

File: software/examples/vbnet/ExampleCallback.vb
Imports System
Imports Tinkerforge

Module ExampleCallback
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Joystick Bricklet

    ' Callback subroutine for pressed callback
    Sub PressedCB(ByVal sender As BrickletJoystick)
        Console.WriteLine("Pressed")
    End Sub

    ' Callback subroutine for released callback
    Sub ReleasedCB(ByVal sender As BrickletJoystick)
        Console.WriteLine("Released")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim j As New BrickletJoystick(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Register pressed callback to subroutine PressedCB
        AddHandler j.PressedCallback, AddressOf PressedCB

        ' Register released callback to subroutine ReleasedCB
        AddHandler j.ReleasedCallback, AddressOf ReleasedCB

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module
